File: pulsar/java/src/main/java/com/examples/hello/pulsar/functions/FunctionController.java
package com.examples.hello.pulsar.functions;

import com.examples.hello.pulsar.PulsarExecutors;
import com.examples.hello.pulsar.ServiceController;
import io.micronaut.http.annotation.Controller;
import jakarta.inject.Inject;
import jakarta.inject.Named;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.concurrent.ExecutorService;

@Controller("/function")
@RequiredArgsConstructor
public class FunctionController extends ServiceController<FunctionRunnable> {
  @Inject @Named(PulsarExecutors.FUNCTIONS) @Getter
  ExecutorService executor;
  @Inject
  @Getter
  FunctionRunnable service;
}

File: pulsar/kotlin/src/main/kotlin/com/example/hello/pulsar/functions/FunctionController.kt
package com.example.hello.pulsar.functions

import com.example.hello.pulsar.PulsarExecutors
import com.example.hello.pulsar.ServiceController
import io.micronaut.http.annotation.Controller
import jakarta.inject.Inject
import jakarta.inject.Named
import java.util.concurrent.ExecutorService

@Controller(value = "/function")

class FunctionController(
  @Inject @Named(PulsarExecutors.FUNCTIONS) val executor: ExecutorService,
  @Inject val service: FunctionRunnable
) : ServiceController<FunctionRunnable>(executor, service)

